streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -265,7 +265,7 @@ private void verifyAllKVKeys(final StreamRunnable[] streamRunnables, final Kafka
                 public boolean conditionMet() {
                     try {
                         final StreamsMetadata metadata = streams.metadataForKey(storeName, key, new StringSerializer());
-                        if (metadata == null) {
+                        if (metadata == null || metadata.equals(StreamsMetadata.NOT_AVAILABLE)) {
```

Comment:
@enothereska @ijuma @dguy OMG I love public API discussions :P

For `store name not exist`: this is not transient since the store names are determined by the time the topology is completely defined; i.e. if a store name does not exist now it will never exist in the future, in this case both throwing an exception (to be more restrictive) or returning null (like in a hash table) looks good to me.

For `metadata not available yet`: this is a transient state, and users should in this case just back-off and retry. If we decided to throw exception in the first case, then perhaps we can just return `null` in this case which do not users to learn about the sentinel value; if we decide to return `null` in the first case then in order to differentiate I'd suggest using a sentinel value.

In general I'd prefer to now throw exceptions for "retry scenarios" unless there is no other choices. Since the `try-catch` pattern looks worse than `if (null)` condition. But again this is purely subjective.